src/main/java/com/apuope/apuope_re/services/LogInService.java

```diff
@@ -0,0 +1,28 @@
+package com.apuope.apuope_re.services;
+
+import com.apuope.apuope_re.entities.User;
+import com.apuope.apuope_re.repositories.UserRepository;
+import com.apuope.apuope_re.dto.UserCredentials;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+
+@Service
+public class LogInService {
+    @Autowired
```

Comment:
I would delete auto wired and create a class constructor instead:

```
@Service
public class LogInService {
    private final UserRepository userRepository;

    public LogInService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

```

Reply:
Fixed